dao/src/main/java/org/thingsboard/server/dao/firmware/BaseFirmwareService.java

```diff
@@ -316,4 +324,12 @@ protected void removeEntity(TenantId tenantId, FirmwareInfo entity) {
             return Optional.empty();
         }
     }
+
+    private static List<FirmwareId> toFirmwareInfoKey(FirmwareId firmwareId) {
+        return Collections.singletonList(firmwareId);
+    }
+
+    private static String toFirmwareDataKey(FirmwareId firmwareId) {
+        return firmwareId.toString();
```

Comment:
The logic of clearing the cache is not obvious. It depends on the implementation of AbstractRedisFirmwareCache.toFirmwareCacheKey? This is not obvious.